### PR TITLE
poco fix fork

### PR DESCRIPTION
**Pull request by paulo-coutinho**
Specify library name and version:  **poco/1.10.1**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/poco/all/conanfile.py [hunk 1]
```diff
@@ -200,6 +202,9 @@ def _configure_cmake(self):
         # On Windows, Poco needs a message (MC) compiler.
         with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
             self._cmake.configure(build_dir=self._build_subfolder)
+        # Disable fork
+        if not self.options.get_safe("enable_fork", True):
+            self._cmake.definitions["POCO_NO_FORK_EXEC"] = True
```
Comment (madebr) [inline recipes/poco/all/conanfile.py:207]:
```suggestion
        # Disable fork
        self._cmake.definitions["POCO_NO_FORK_EXEC"] = not self.options.get_safe("enable_fork", True)
```

File: recipes/poco/all/conanfile.py [hunk 2]
```diff
@@ -21,10 +21,12 @@ class PocoConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "enable_fork": [True, False],
```
Comment (madebr) [inline recipes/poco/all/conanfile.py:24]: `fork` is not available on Windows.
So I think it should be conditionally removed in `config_options`.

**Comment by paulo-coutinho** [comment]
Hi,

On Windows is ignored because the file that have this condition is on Linux.

Is necessary remove this option on conanfile?

**Comment by madebr** [comment]
It's nice to not have two identical packages on Windows with different package ids.
This is the same use case as for removing fpic on Windows

**Comment by paulo-coutinho** [comment]
> It's nice to not have two identical packages on Windows with different package ids.
> This is the same use case as for removing fpic on Windows

Done

**Comment by paulo-coutinho** [comment]
Hi,

The error was:

```
MSBUILD : error MSB4166: Child node "5" exited prematurely. Shutting down. Diagnostic information may be found in files in "C:\J\w\BuildSingleReference@12\20971\763181e6-6acb-41ed-93de-259d5be53e88\" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.
```

It was related to CI server or to recipe?

**Comment by madebr** [comment]
The boost recipe also got a lot of these kind of errors.
I fixed/worked around them by adding `cmake.parallel = False` to its test_package.
https://github.com/conan-io/conan-center-index/blob/e1d4229393d02ab88444ea9e59f25eb60169890c/recipes/boost/all/test_package/conanfile.py#L41-L42

**Comment by paulo-coutinho** [comment]
> The boost recipe also got a lot of these kind of errors.
> I fixed/worked around them by adding `cmake.parallel = False` to its test_package.
> https://github.com/conan-io/conan-center-index/blob/e1d4229393d02ab88444ea9e59f25eb60169890c/recipes/boost/all/test_package/conanfile.py#L41-L42

Hi,

Ok. I need add this here too?

Thanks.

**Comment by madebr** [comment]
The number of errors or times you need to restart the build will be lower.

**Comment by madebr** [comment]
Looks like the test package build gets killed, even with parallel builds disabled..

**Comment by jgsogo** [comment]
We have added `MSBUILDDISABLENODEREUSE=1` to Windows environment. It might help with these errors.

File: recipes/poco/all/conanfile.py [hunk 3]
```diff
@@ -200,11 +203,16 @@ def _configure_cmake(self):
...
+        # Disable fork
+        if not self.options.get_safe("enable_fork", True):
+            self._cmake.definitions["POCO_NO_FORK_EXEC"] = True
         return self._cmake
 
     def build(self):
         self._patch_sources()
         cmake = self._configure_cmake()
+        # Disable parallel builds because c3i (=conan-center's test/build infrastructure) seems to choke here
+        cmake.parallel = False
```
Comment (jgsogo) [inline recipes/poco/all/conanfile.py:215]: I think we can revert this change. The error came from the `test_package`, not from the library build... so, if needed, it should go in the `test_package/conanfile.py` (but please, let's try it first now that we have added the `MSBUILDDISABLENODEREUSE` envvar)

**Comment by paulo-coutinho** [comment]
Finally!

File: recipes/poco/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/poco/all/conanfile.py:208]: Incorporating this suggestion would have been nice, but I won't push since getting this one though c3i is a pain.

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]
